src/ipc/handler.c

```diff
@@ -677,6 +677,13 @@ static int ipc_pm_gate(uint32_t header)
 	else
 		pm_runtime_enable(PM_RUNTIME_DSP, PLATFORM_MASTER_CORE_ID);
 
+#if !CONFIG_CAVS_LPRO
+	if (pm_runtime_is_active(PM_RUNTIME_DSP, PLATFORM_MASTER_CORE_ID))
+		platform_set_active_clock(CPU_HPRO_FREQ_IDX);
+	else
+		platform_set_active_clock(CPU_LPRO_FREQ_IDX);
+#endif
```

Comment:
@mmaka1 @slawblauciak we also send the PM_GATE IPC when the DSP enters D0i3 in S0. The only way  for the FW to know if it is sent during S0 or S0ix is by checking whether the DMA_TRACE is requested to be turned off or not. Do you think it makes a difference whether the switching to LPRO is happening during S0 as well?